controllers/storagecluster/cephcluster.go

```diff
@@ -1529,9 +1530,17 @@ func isEncrptionSettingUpdated(clusterWideEncrytion bool, existingDeviceSet []ro
 }
 
 // setDefaultMetadataPoolSpec sets the common pool spec for all metadata pools as necessary
-func setDefaultMetadataPoolSpec(poolSpec *rookCephv1.PoolSpec, sc *ocsv1.StorageCluster) {
-	poolSpec.EnableCrushUpdates = true
-	if poolSpec.DeviceClass == "" {
+func (r *StorageClusterReconciler) setDefaultMetadataPoolSpec(poolSpec *rookCephv1.PoolSpec, sc *ocsv1.StorageCluster, specifiedPoolSpecPath []string) {
+	// Check if enableCrushUpdates is explicitly set in the unstructured obj else set to true
+	if val, found := getValueAtPath(r.unstructuredSC, append(specifiedPoolSpecPath, "enableCrushUpdates")); found {
+		poolSpec.EnableCrushUpdates = val.(bool)
+	} else {
+		poolSpec.EnableCrushUpdates = true
+	}
+	// Check if DeviceClass is explicitly set in the unstructured obj else set to the default device class
+	if val, found := getValueAtPath(r.unstructuredSC, append(specifiedPoolSpecPath, "deviceClass")); found {
+		poolSpec.DeviceClass = val.(string)
+	} else if poolSpec.DeviceClass == "" {
```

Comment:
Is it possible for the user to explicitly set blank? Seems like we need to still enforce the default device class in that case.
```suggestion
	}
	if poolSpec.DeviceClass == "" {
```

Reply:
Okay, Thanks for the clarification removed the check for deviceClass now